nixnet/database/_dbc_attributes.py

```diff
@@ -0,0 +1,99 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import collections
+import six
+import typing  # NOQA: F401
+
+from nixnet import _funcs
+from nixnet import constants
+
+
+class DbcAttributeCollection(collections.Mapping):
+
+    def __init__(self, handle):
+        # type: (int) -> None
+        self._handle = handle
+
+    def __repr__(self):
+        return 'DbcAttributeCollection(handle={0})'.format(self._handle)
+
+    def __eq__(self, other):
+        if isinstance(other, self.__class__):
+            return self._handle == other._handle
+        else:
+            return NotImplemented
+
+    def __ne__(self, other):
+        result = self.__eq__(other)
+        if result is NotImplemented:
+            return result
+        else:
+            return not result
+
+    def __hash__(self):
+        return hash(self._handle)
+
+    def __len__(self):
+        return len(list(self.keys()))
+
+    def __iter__(self):
+        return self.keys()
+
+    def __getitem__(self, index):
+        if isinstance(index, six.string_types):
+            return self._value(index)
+        else:
+            raise TypeError(index)
+
+    def keys(self):
+        mode = constants.GetDbcAttributeMode.ATTRIBUTE_LIST
+        attribute_size = _funcs.nxdb_get_dbc_attribute_size(self._handle, mode, '')
+        attribute_info = _funcs.nxdb_get_dbc_attribute(self._handle, mode, '', attribute_size)
+        name_string = attribute_info[0]
+        name_list = []  # type: typing.List[typing.Text]
+        if len(name_string):
+            name_list = name_string.split(',')
+        for name in name_list:
```

Comment:
No need to create a list. Can be simplified to:
`for name in name_string.split(",")`
  `if name.strip():`